api/api/services/internal_tasks/agent_creation_service.py

```diff
@@ -0,0 +1,165 @@
+    OpenAIProxyChatCompletionRequest,
+    OpenAIProxyChatCompletionResponse,
+    OpenAIProxyMessage,
+    OpenAIProxyResponseFormat,
+)
+from api.services.feedback_svc import FeedbackTokenGenerator
+from api.services.groups import GroupService
+from api.services.run import RunService
+from core.agents.agent_creation_agent import CreateAgentToolCall, agent_creation_agent
+from core.domain.events import EventRouter
+from core.domain.fields.chat_message import ChatMessage
+from core.domain.tenant_data import TenantData
+from core.storage.backend_storage import BackendStorage
+
+
+class AgentCreationResult(BaseModel):
+    agent_id: str
+    agent_schema_id: int
+    version_id: str
+    run_id: str
+
+
+class AgentCreationChatResponse(BaseModel):
+    assistant_answer: str
+    agent_creation_result: AgentCreationResult | None
+
+
+class AgentCreationService:
+    def __init__(self, storage: BackendStorage, event_router: EventRouter):
+        self.logger = logging.getLogger(self.__class__.__name__)
+        self.storage = storage
+        self._event_router = event_router
+
+    async def _handle_agent_creation(
+        self,
+        user_org: TenantData,
+        group_service: GroupService,
+        run_service: RunService,
+        event_router: EventRouter,
+        feedback_generator: FeedbackTokenGenerator,
+        agent_creation_tool_call: CreateAgentToolCall,
+    ):
+        # TODO: plug actual data from agent tool call
+
+        handler = OpenAIProxyHandler(
+            group_service=group_service,
+            storage=self.storage,
+            run_service=run_service,
+            event_router=event_router,
+            feedback_generator=feedback_generator,
+        )
+
+        messages = [
+            OpenAIProxyMessage(
+                role="system",
+                content=agent_creation_tool_call.system_message_content,
+            ),
+        ]
+        if agent_creation_tool_call.user_message_content:
+            messages.append(
+                OpenAIProxyMessage(
+                    role="user",
+                    content=agent_creation_tool_call.user_message_content,
+                ),
+            )
+
+        # Create a completion request that will be executed
+        completion_request = OpenAIProxyChatCompletionRequest(
+            model="gpt-4o-latest",
+            messages=messages,
+            agent_id=agent_creation_tool_call.agent_name,
+            stream=False,
+        )
+
+        if agent_creation_tool_call.response_format:
+            completion_request.response_format = OpenAIProxyResponseFormat(
+                type="json_schema",
+                json_schema=OpenAIProxyResponseFormat.JsonSchema(
+                    schema=agent_creation_tool_call.response_format,
+                ),
+            )
+
+        prepared_run = await handler.prepare_run(completion_request, user_org)
+
+        response = await handler.handle_prepared_run(
```

Comment:
@guillaq, as part of the new agent creation flow, when the agent decide to create an agent, I actually create the agent and make a run using the `OpenAIProxyHandler`. This is the quickest way I found to get the job done.

Also, @jacekzimonski told me the best thing for him would be to have a `run_id` and by doing that I can provide a run id to him.

Does the method look all right to you ?